### PR TITLE
bug fix for AR date

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#6197

*Description of changes:*
- get AR year from AR Filing Year if available, otherwise get year from Annual Report Date as before

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by cameron-freshworks — APPROVED** [review]
LGTM!

File: src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -475,46 +475,43 @@ export default {
       const annualReport = filing?.annualReport
 
       if (header && annualReport) {
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:477]: Use Ignore whitespace changes option to view this please.

File: src/components/Dashboard/FilingHistoryList.vue [hunk 2]
```diff
@@ -475,46 +475,43 @@ export default {
...
-        const date = annualReport.annualReportDate
-        if (date) {
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:479]: We will always get a date here, so check is not needed.

File: src/components/Dashboard/FilingHistoryList.vue [hunk 3]
```diff
@@ -475,46 +475,43 @@ export default {
...
-              corpName: this.entityName || this.entityTypeToNumberedDescription(this.entityType),
-              filingDateTime,
-              paymentToken: header.paymentToken,
-              title: 'Receipt',
-              filename: `${this.getEntityIncNo} - Receipt - ${filingDate}.pdf`
-            })
-          }
+        // get AR Filing Year from header if available
+        // otherwise get year from Annual Report Date
+        const agmYear = header.ARFilingYear || +annualReport.annualReportDate.slice(0, 4)
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:482]: AR Filing Year is a new field -- need fallback as older ARs don't have it!

**Pull request merged**